docs/internal/generics.md

```diff
@@ -0,0 +1,114 @@
+# Generics
+
+Generics are a way to use type parameters to define or restrict other types.
+For example, a generic class can use its type parameters to define the type of fields, and a generic method can use its parameters to restrict what can be passed as an argument.
```

Comment:
```suggestion
For example, a generic class can use its type parameters to define the type of fields, and a generic method can use its parameters to provide specialisations for different argument types.
```

I'm not really happy with this suggested change, but non-generic types also restrict what can be passed as an argument so it's difficult to understand what generics add with that phrasing.